source/mpi_fsi.cpp

```diff
@@ -582,25 +582,26 @@ namespace MPI
                                    vertices_mask);
                     interpolator.point_value(fluid_solver.present_solution,
                                              value);
-                    std::vector<Tensor<1, dim>> gradient(dim + 1,
-                                                         Tensor<1, dim>());
-                    interpolator.point_gradient(fluid_solver.present_solution,
-                                                gradient);
-                    // Compute stress
-                    SymmetricTensor<2, dim> sym_deformation;
-                    for (unsigned int i = 0; i < dim; ++i)
+                    Utils::GridInterpolator<dim, PETScWrappers::MPI::Vector>
+                      scalar_interpolator(fluid_solver.scalar_dof_handler,
+                                          s_cell->face(f)->vertex(v),
+                                          vertices_mask);
+                    SymmetricTensor<2, dim> viscous_stress;
+                    for (unsigned int i = 0; i < 2; i++)
                       {
-                        for (unsigned int j = 0; j < dim; ++j)
+                        for (unsigned int j = 0; j < i + 1; j++)
```

Comment:
Think this is not correct. In case of ``dim = 2``,  ``s_00 s_01 s_10 s_11 s_12`` will be evaluated, but index 2 doesn't exist, which causes undefined behavior. The correct form should be:
```c++
for (unsigned j = i; j < dim; ++j)
{
}
```